semgrep-core/src/tainting/Taint.ml

```diff
@@ -30,7 +30,7 @@ type 'a call_trace =
 
 type source = Rule.taint_source call_trace [@@deriving show]
 type sink = Rule.taint_sink call_trace [@@deriving show]
-type arg_pos = int [@@deriving show]
+type arg_pos = string * int [@@deriving show]
```

Comment:
We should probably rename arg_pos to arg then and use a variant also here?

Reply:
You mean, do you want a unary constructor for the `arg` type?

Reply:
I mean in other places you've changed the argument type to change from a simple expression to Unamed | Named.
Now in other code we used to have a type to record the position of an argument (an int), and now we don't care just about position so arg_pos seems like a bad name. Or maybe I don't understand what the string and int mean in this arg_pos type in which can a comment would help clarify.

Reply:
I'm not even sure about the `string * int` type, maybe
```ocaml
type arg = Pos of int | Named of string
```

Reply:
That is what it started out as. It seems like you would want a sum here, but actually you don't. This isn't going to work because every argument (in Python, at least) can be used both positionally and in a named fashion. If the function is invoked with that positional information, then you need to know which `arg_pos` to look for based on index. If the function is invoked with named information, then you need to know which `arg_pos` to look for based on name.